lib/StreamingStore.ts

```diff
@@ -1,4 +1,4 @@
-import type { EventEmitter } from 'events';
+import { EventEmitter } from 'events';
```

Comment:
Maybe using `EventEmitter` here is a bit overkill (especially since we'd need to add `'events'` as an actual dependency now), since we only really need a single type of event.
Maybe we can just implement `addEndListener` on the `StreamingStore` class?